httpgrpc/server/server_test.go

```diff
@@ -326,3 +327,125 @@ func TestTracePropagation(t *testing.T) {
 	assert.Equal(t, "world", recorder.Body.String())
 	assert.Equal(t, 200, recorder.Code)
 }
+
+func TestGrpcErrorsHaveCorrectMessage(t *testing.T) {
+	testCases := map[string]struct {
+		responseBody         string
+		errorMessageInHeader string
+
+		expectedErrorMessage string
+	}{
+		"error response with string body": {
+			responseBody:         "hello world",
+			expectedErrorMessage: "rpc error: code = Code(500) desc = hello world",
+		},
+		"error response with binary body": {
+			responseBody:         "\x08\x08\x12\xc7\x03the request has been rejected",
+			expectedErrorMessage: "rpc error: code = Code(500) desc = \x08\x08\x12\xc7\x03the request has been rejected",
+		},
+		"error response with binary body and provided message via header": {
+			responseBody:         "\x08\x08\x12\xc7\x03the request has been rejected",
+			errorMessageInHeader: "hello world",
+			expectedErrorMessage: "rpc error: code = Code(500) desc = hello world",
+		},
+	}
+	for testName, testData := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if testData.errorMessageInHeader != "" {
+					w.Header().Set(ErrorMessageHeaderKey, testData.errorMessageInHeader)
+				}
+				w.WriteHeader(http.StatusInternalServerError)
+				_, _ = w.Write([]byte(testData.responseBody))
+			})
+
+			s := NewServer(h)
+			req := &httpgrpc.HTTPRequest{Method: "GET", Url: "/test"}
+			resp, err := s.Handle(context.Background(), req)
+			require.NotNil(t, err)
+			require.Nil(t, resp)
+
+			require.Equal(t, testData.expectedErrorMessage, err.Error())
+
+			httpResp, ok := httpgrpc.HTTPResponseFromError(err)
+			require.True(t, ok)
+			// Verify that header was removed
+			require.Empty(t, httpResp.Headers)
+		})
+	}
+}
+
+func Test(t *testing.T) {
```

Comment:
Is this copy-pasted from `TestGrpcErrorsHaveCorrectMessage` by mistake?

Reply:
lol, yes. no idea how that happened :) thanks for spotting!